### PR TITLE
Allow numpy array passed into autothresholding function

### PR DESCRIPTION
**Pull request by yrkim98**
## What does this PR do?
Allows numpy arrays to be passed into the auththresholding function in `AutoThresh.py` while still allowing torch tensors as before.

Allows thresholding to work properly

### CODE REVIEW

File: cyto_dl/models/im2im/utils/postprocessing/auto_thresh.py [hunk 1]
```diff
@@ -23,7 +23,10 @@ def thresh_func(image):
         self.thresh_func = thresh_func
 
     def __call__(self, image):
-        image = image.detach().cpu().float().numpy()
+        # Only get numpy array from torch tensor if tensor is passed in
+        # Allows np.ndarray to be passed in directly
+        if not isinstance(image, np.ndarray):
```
Comment (hughes036) [inline cyto_dl/models/im2im/utils/postprocessing/auto_thresh.py:28]: Im not a fan of reassigning an arg (I know this was already happening before the change).  Could you use a new local var?

File: cyto_dl/models/im2im/utils/postprocessing/auto_thresh.py [hunk 2]
```diff
@@ -23,7 +23,10 @@ def thresh_func(image):
...
+            image = image.detach().cpu().float().numpy()
```
Comment (benjijamorris) [inline cyto_dl/models/im2im/utils/postprocessing/auto_thresh.py:29]: (feel free to ignore since this is out of the scope of the pr)
there's also a [detach]("\\allen\aics\assay-dev\users\Benji\CurrentProjects\im2im_dev\cyto-dl\logs\train\runs\diffae\orig_model\2024-12-06_12-47-05\checkpoints\last.ckpt") function that does the detach->cpu->numpy conversion, so this could be replaced with 
`img = detach(img).astype(float)`

File: cyto_dl/models/im2im/utils/postprocessing/auto_thresh.py (hunk 1 above)
Comment (benjijamorris) [inline cyto_dl/models/im2im/utils/postprocessing/auto_thresh.py:28]: (same as below) this might be better as `if isinstance(image, torch.Tensor)`

**Review by benjijamorris — APPROVED** [review]

**Review by hughes036 — APPROVED** [review]
